Sources/TensorFlow/Initializers.swift

```diff
@@ -509,17 +531,28 @@ public extension Tensor where Scalar: BinaryFloatingPoint,
     }
 }
 
+//===------------------------------------------------------------------------------------------===//
+// Variance Scaling
+//===------------------------------------------------------------------------------------------===//
+
 fileprivate extension Tensor where Scalar: TensorFlowFloatingPoint {
-    private static func glorot(
-        fromStandardUniform randomUniform: __shared Tensor<Scalar>,
-        shape: __shared TensorShape
-    ) -> Tensor<Scalar> {
-        let spatialDimCount = shape.count - 2
-        let receptiveField = shape[0..<spatialDimCount].contiguousSize
-        let fanIn = shape[shape.count - 2] * receptiveField
-        let fanOut = shape[shape.count - 1] * receptiveField
-        let minusOneToOne = 2 * randomUniform - 1
-        return Scalar.sqrt(Scalar(6) / Scalar(fanIn + fanOut)) * minusOneToOne
+    // Returns the input and output channel counts, `(fanIn, fanOut)`, of a given tensor shape.
+    private static func computeFans(shape: TensorShape) -> (Int, Int) {
```

Comment:
- According to the Swift API Design Guidelines, non-side-effecting methods should read like a noun. I'd suggest renaming this to `fans(shape:) -> (in: Int, out: Int)`.
- Compensate for weak type information in the result type.
```suggestion
    private static func fans(shape: TensorShape) -> (in: Int, out: Int) {
```